### PR TITLE
Feature/precommit

### PR DESCRIPTION
**Pull request by thomasburgess**
Added pre-commit support, ran chain and fixed issues. This changes pretty much every file in the repo, but subsequent commits using pre-commit should be more consistent. Most changes are minor, but a few issues appeared in Comparer/Regressor.

### CODE REVIEW

**Comment by thomasburgess** [comment]
Just FYI, pre commit does this
* Basics
  * Check python code parses
  * Check file name case conflict
  * Check doc string first
  * Check executables have shebangs, i.e. #!/bin/sh
  * Check for presence of merge conflict <<<<< ==== stuff
  * Check toml and yaml file syntax
  * Check for debug statements
  * Ensure files ends on new line and trim white space
* Black - format code
* isort - sort import statements
* mypy - perform type checking
* flake8 - lint code

File: methcomp/glucose.py [hunk 1]
```diff
@@ -807,124 +1251,163 @@ def __init__(self, reference, test, units,
...
 
         return _zones
 
     def plot(self, ax):
         # ref, pred
-        ref = self.reference
-        pred = self.test
-        from . import static # temporary fix 
-        _data = np.loadtxt(pkg_resources.open_text(static, 'seg.csv'))
+        #        ref = self.reference NOT USED
```
Comment (thomasburgess) [inline methcomp/glucose.py:1294]: NOTE: flake8 complained about not used, I don't understand what it does, but it is now commented out

File: methcomp/glucose.py [hunk 2]
```diff
@@ -807,124 +1251,163 @@ def __init__(self, reference, test, units,
...
+        #        pred = self.test NOT USED
+        from . import static  # temporary fix
 
+        #        _data = np.loadtxt(pkg_resources.open_text(static, "seg.csv"))
```
Comment (thomasburgess) [inline methcomp/glucose.py:1298]: NOTE: flake8 complained about not used

File: methcomp/glucose.py [hunk 3]
```diff
@@ -807,124 +1251,163 @@ def __init__(self, reference, test, units,
...
+        )
 
         # Plot color axes
-        from . import static # temporary fix
-        grid_path = str(path_func(static, 'seg600.png'))
-        cax = ax.imshow(np.flipud(np.array(plt.imread(grid_path))),
-                        origin='lower',
-                        cmap=cmap,
-                        vmin=0, vmax=4)
+        # from . import static  # temporary fix # ALREADY DONE
```
Comment (thomasburgess) [inline methcomp/glucose.py:1320]: NOTE: flake8 complained about already imported

**Review by thomasburgess — COMMENTED** [review]
Some comments on glucose

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -0,0 +1,36 @@
+
+repos:
```
Comment (wptmdoorn) [inline .pre-commit-config.yaml:2]: I've read up about pre-commit-config files and these all look OK but I'm not an expert on this subject.

**Review by wptmdoorn — COMMENTED** [review]
Looks good. It seems like a lot of changes but I agree that this is mainly due to the style being adjusted now for once.

**Comment by thomasburgess** [comment]
Yes, most changes are just moving whitespace around... but later on whitespace changes will be minimized instead...

File: methcomp/glucose.py (hunk 1 above)
Comment (wptmdoorn) [reply methcomp/glucose.py:1294]: this is actually correct, the functionality got moved to `.calc_error_score`. If you want, you can remove it for now, but when I will implement the Comparer functionality for the Surveillance Error Grids this is gonna be re-done anyhow.

**Comment by thomasburgess** [comment]
If i didn’t break, we keep it as is...

File: methcomp/glucose.py (hunk 2 above)
Comment (wptmdoorn) [reply methcomp/glucose.py:1298]: this is also correct, can be removed but will also be re-done in a new version using the Comparer class

**Comment by wptmdoorn** [comment]
Looks very good, do you want to perform the changes in the glucose file or not? Otherwise I would say we are ready to merge this branch.

File: methcomp/glucose.py (hunk 2 above)
Comment (thomasburgess) [reply methcomp/glucose.py:1298]: We keep this version for now, and can do a more indepth review and fixing when we re-do stuff anyway. As long as I didn't break it I'm happy at this stage :)

**Comment by thomasburgess** [comment]
Let's go ahead and merge, then we are in a good place to to make Glucose and BlandAltman comparers

**Pull request merged**